src/elf/sym.rs

```diff
@@ -186,20 +187,11 @@ macro_rules! elf_sym_std_impl {
                 }
             }
 
-            pub unsafe fn from_raw<'a>(symp: *const Sym, count: usize) -> &'a [Sym] {
-                slice::from_raw_parts(symp, count)
-            }
-
             pub fn from_fd<'a>(fd: &mut File, offset: usize, count: usize) -> Result<Vec<Sym>> {
                 // TODO: AFAIK this shouldn't work, since i pass in a byte size...
-                // FIX THIS, unecessary allocations + unsafety here
-                let mut bytes = vec![0u8; count * SIZEOF_SYM];
+                let mut syms = vec![Sym::default(); count];
                 try!(fd.seek(Start(offset as u64)));
-                try!(fd.read(&mut bytes));
-                let bytes = unsafe { slice::from_raw_parts(bytes.as_ptr() as *mut Sym, count) };
-                let mut syms = Vec::with_capacity(count);
-                syms.extend_from_slice(bytes);
-                syms.dedup();
```

Comment:
Also, from_raw definition above. That one was deliberate. There's no reason to provide such violently unsafe functions, especially if it's just an alias for core function.